### DISCUSSION TITLE
hidden some item in repeater based on condition

### DISCUSSION THREAD

**Discussion by jechazelle** [filamentphp/filament#9862] [Help]
Hi,

I would like to know how can I hidden some items in repeater field based on condition.

For example, I would like to hide the items when language_uuid not matching.

I tried it:

```
Repeater::make('translations')
  ->label('')
  ->relationship()
  ->schema([
      Group::make()
          ->schema([
              Group::make()
                  ->schema([
                      Toggle::make('is_active')
                        ->label('Active')
                        ->inline(false),
                  ])->columns(3),
              RichEditor::make('one_sentence')
                  ->label('Contenu'),
              ])->visible(fn ($state) => self::checkIfUserLanguageMatchWithTranslationLanguage($state)),
  ])

---


public static function checkIfUserLanguageMatchWithTranslationLanguage($state)
{
    return Auth::getUser()->getPreferenceLanguageUuid() === $state['language_uuid'];
}

```

But I have an items blank:

<img width="587" alt="Capture d’écran 2023-11-25 à 15 09 52" src="https://github.com/filamentphp/filament/assets/6713790/6701cd8f-f0d1-4f1f-9edd-47af3bfdda29">

I would like only the second item in this example and hide completly the first and the last.

Thank you for your help!

**Comment by leandrocfe** [comment]
Where is the `language_uuid`? If you want to get a value outside the repeater, you should use `$get('../../language_uuid')`
https://filamentphp.com/docs/3.x/forms/fields/repeater#using-get-to-access-parent-field-values

```php
->visible(fn (Get $get) => self::checkIfUserLanguageMatchWithTranslationLanguage($get('../../language_uuid')))
```
You need to add `->live()` to the `language_uuid` field

**Reply by jechazelle** [reply]
Thx @leandrocfe for your answer.

language_uuid is on the item.

Currentlye I have a post with post_translations

post : uuid
post_translations : uuid, language_uuid, one_sentence

The repeater iterate on each translation.

I would like to hide the item translation (in repeater) when language_uuid isn't getPreferenceLanguageUuid from user model.
I would like to hide a specific repeater index when not match.

<img width="441" alt="Capture d’écran 2023-11-27 à 16 19 50" src="https://github.com/filamentphp/filament/assets/6713790/d2ad9ec0-da27-4d7e-8496-b6c9a7981bdb">

That's make sense ?

**Reply by leandrocfe** [reply]
Hum... not sure if you can do it
